listenbrainz/webserver/static/js/src/playlists/Playlist.tsx

```diff
@@ -568,6 +579,7 @@ export default class PlaylistPage extends React.Component<
         currentUser.auth_token
       );
       this.setState({ playlist: JSPFObject.playlist });
+      this.emitPlaylistChanged();
```

Comment:
We should move this line up to right after we get confirmation that the API call succeeded, Line 571 right after `this.newAlert("success", "Saved playlist", "");`.

That way we correctly emit the change even if something goes wrong in this trycatch block.

And behold! Magic happens: we don't actually need this trycatch block at all and we can remove it!
Sicne we get updates through websockets, this whole `Fetch the newly editd playlist and save it to state` is redundant.

Reply:
I see you removed the call to `emitPlaylistChanged`, but that's not what I meant, sorry if I was not clear.

You should indeed call `emitPlaylistChanged` right after `this.newAlert("success", "Saved playlist", "")` and remove the block of code that comes after that fetches the playlist through the API: https://github.com/metabrainz/listenbrainz-server/pull/1212/files#diff-4a293c4781b3b1e784e1cad81c20cd70c3c01f8d7ef9526aab4294126eeeea9eR580-R589

We don't need to call the API anymore since we receive the updated playlist via WS